csharp/CacheOption.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Runtime.InteropServices;
+
+namespace ParquetSharp
+{
+    public struct CacheOption
```

Comment:
This should be called `CacheOptions` to match the C++ library and because it contains multiple options.